packages/realm-server/tests/permissions/permission-checker-test.ts

```diff
@@ -21,6 +23,30 @@ module('realm-user-permissions', function (_hooks) {
     test('anyone can read and write', function (assert) {
       assert.ok(permissionsChecker.can('anyone', 'read'));
       assert.ok(permissionsChecker.can('anyone', 'write'));
+
+      assert.deepEqual(permissionsChecker.for('anyone'), ['read', 'write']);
+    });
+  });
+
+  module('users-readable realm', function () {
+    let permissionsChecker = new RealmPermissionChecker({
+      users: ['read'],
+      '@matic:boxel-ai': ['read', 'write'],
+    });
+
+    test('user with permission can read but not write', function (assert) {
+      assert.ok(permissionsChecker.can('@matic:boxel-ai', 'read'));
+      assert.ok(permissionsChecker.can('@matic:boxel-ai', 'write'));
+
+      assert.deepEqual(permissionsChecker.for('@matic:boxel-ai'), [
+        'read',
+        'write',
+      ]);
+
+      assert.ok(permissionsChecker.can('anyone', 'read'));
```

Comment:
this seems wrong. the permissions are not world readable, only matrix users should be able to read. `anyone` does not seem to be a valid matrix username